src/teehr/metrics/gumboot_bootstrap.py

```diff
@@ -0,0 +1,185 @@
+"""Bootstrap classes."""
+from typing import Any, Callable, cast, Union
+from collections.abc import Generator as PyGenerator
+from pathlib import Path
+import logging
+
+from arch.bootstrap import IIDBootstrap
+from arch.bootstrap.base import (
+    # _loo_jackknife, # leave one out jackknife
+    _add_extra_kwargs
+)
+from arch.typing import ArrayLike, Int64Array, Float64Array, NDArray
+from numpy.random import Generator, RandomState
+import numpy as np
+import pandas as pd
+
+logger = logging.getLogger(__name__)
+
+# FILL_VALUE = -9999  # JAB
+
+
+class GumbootBootstrap(IIDBootstrap):
+    """Python implementation of the Gumboot R package."""
+
+    def __init__(
+        self,
+        *args: ArrayLike,
+        value_time: pd.Series,
+        water_year_month: Union[int, None] = None,
+        boot_year_file: Union[str, Path, None] = None,
```

Comment:
I'm a little unsure about passing in a file path here.  Lets discuss.

Reply:
Sure.  I'm open to suggestions here, including removing the option for a boot year file.  It's kind of helpful for testing but it's hard to imagine users specifying their own.